packages/table/preview/index.tsx

```diff
@@ -306,15 +308,37 @@ class MutableTable extends React.Component<{}, IMutableTableState> {
     }
 
     private renderColumnHeaderCell = (columnIndex: number) => {
+        let renderName: (name: string) => JSX.Element;
+        if (this.state.enableColumnCustomHeaders) {
+            renderName = this.renderCustomColumnName;
+        } else if (this.state.enableColumnNameEditing) {
+            renderName = this.renderEditableColumnName;
+        } else {
+            renderName = undefined;
+        }
```

Comment:
prefer helper function with several `return` statements instead of `let` variable